### PR TITLE
MM-53058: Move cache layer out of timer layer

### PR DESCRIPTION
**Pull request by agnivade**
Having cache layer as part of the store call
Prometheus metrics can be somewhat misleading
because cache-backed methods can have very high
volume and yet not cause any load on the DB at all.

However, if the admin is not aware which calls
are cache-backed and which are not, then
it can be hard to understand why a store call
of high volume isn't causing any stress on the DB at all.

Therefore, for simplicity we reorder the layers
such that cache layer is handled outside the timer layer.

We already have cache related metrics like hit/miss
which can be used to monitor cache usage.

https://mattermost.atlassian.net/browse/MM-53058

```release-note
NONE
```

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]

**Comment by agnivade** [comment]
@saturninoabril - The link to the test dashboard is not opening anything. Is there something wrong?

**Comment by saturninoabril** [comment]
Thanks for reporting. This should be good now.

**Comment by agnivade** [comment]
Do I need to kick off a new test, or will this one start again? It says "on-queue"

**Comment by saturninoabril** [comment]
No need to kick off the new test. I restarted the job and is in progress testing.

**Comment by DHaussermann** [comment]
Regarding the test failures of interest - none seem relevant to the code change.  

- T329 is failing because the image in the test hosted at `https://raw.githubusercontent.com/furqanmlk/furqanmlk.github.io/main/images/image-small-height.png` returns a 404
- T331 is also failing for the same image as above 
- T668 is **not** failing when I try manually on the PR server here 
- T669 is also working fine with Manual tests

Of the remaining 3 failures (not sure exactly which Zephyr test they map to) none fail when testing with manually with random data. and 2 of the 3 tests fail because they are also trying to have image assets served from some permutation of `https://raw.githubusercontent.com/furqanmlk/furqanmlk.github.io`

In summary: Most failures are caused by resources hosted on `https://raw.githubusercontent.com/furqanmlk/furqanmlk.github.io` that seem to no longer be available and perhaps the `/collapse` / `/expand` tests are flaky? 

@saturninoabril not sure what the maintenance process would be here. Tests needing these images were not broken by a specific PR but need maintenance - including test data construction.

**Comment by DHaussermann** [comment]
@agnivade I've done a very quick sanity check with data I built on the server above ☝️ . 

I'm having an issue creating a `/cloud` server that has test data. I would need a server with the pre-canned test data. I'll get this sorted shortly and then complete this.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
I deployed this locally where I was able to populate the standard test data. I then smoke tested this change
- Very basic Mattermost smoke test performing most common actions
- Smoke tested search functionality
  - Results are returned as expected
  - Business logic for results within teams is respected as well as results within publiic and private channels based on channel membership.
  - Search targeting posts in DM and GMs is working and is team agnostic
  - Searched for various terms and word combinations common within the Lorem Ipsum text. Results seem normal   
  - No performance delays noticed when returning  the set of results based on terms
  - All operators for `in`, `from` `before` etc... seem to works as expected
  - When using many operators together to build a very specific query within a channel it can take a second or so longer to look far back in a specific channel to populate the initial set of results results but, I notice nothing out of the ordinary in terms of performance. 
  
No issues found. 

LGTM!

**Comment by agnivade** [comment]
@saturninoabril @mvitale1989 - It looks like some e2e tests are failing. What is the expected action to take here?

**Comment by saturninoabril** [comment]
It requires an update from https://github.com/mattermost/mattermost/pull/23748, specifically the addition of env variable which was missed by the recent change to Insights. I'll update and let's see if this will pass as expected.
